### PR TITLE
Fix PD member name match for multi-cluster when dispatching join parameters in discovery

### PR DESCRIPTION
**Pull request by handlerww**
### What problem does this PR solve?
Fix bug mentioned in https://github.com/pingcap/tidb-operator/pull/3530#discussion_r544741762

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/discovery/discovery.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 		// In some failure situations, for example, delete the pd's data directory, pd will try to restart
 		// and get join info from discovery service. But pd embed etcd may still have the registered member info,
 		// which will return the argument to join pd itself, which is not suggested in pd.
-		if member.Name == podName {
+		if member.Name == podName || member.Name == strArr[0] {
```
Comment (csuzhangxc) [inline pkg/discovery/discovery.go:146]: Can you give an example in the comment here?

Comment (handlerww) [reply pkg/discovery/discovery.go:146]: It corresponds to https://github.com/tikv/pd/blob/43baea981b406df26cd49e8b99cc42354f0a6696/server/join/join.go#L88.
When multi-cluster enabled, the PD member name is not pod name(cluster1-pd-0) but the FQDN (cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local).

Comment (csuzhangxc) [reply pkg/discovery/discovery.go:146]: I mean, you can add this comment in the code (for other developers or even you). 😹

File: pkg/discovery/discovery.go [hunk 2]
```diff
@@ -143,7 +143,7 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 		// In some failure situations, for example, delete the pd's data directory, pd will try to restart
 		// and get join info from discovery service. But pd embed etcd may still have the registered member info,
 		// which will return the argument to join pd itself, which is not suggested in pd.
```
Comment (handlerww) [inline pkg/discovery/discovery.go:145]:
```suggestion
                // corresponds to https://github.com/tikv/pd/blob/43baea981b406df26cd49e8b99cc42354f0a6696/server/join/join.go#L88.
                // When multi-cluster enabled, the PD member name is not pod name(cluster1-pd-0) but the FQDN (cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local).
		// In some failure situations, for example, delete the pd's data directory, pd will try to restart
		// and get join info from discovery service. But pd embed etcd may still have the registered member info,
		// which will return the argument to join pd itself, which is not suggested in pd.
		// 
```

File: pkg/discovery/discovery.go [hunk 3]
```diff
@@ -140,10 +140,13 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 
 	membersArr := make([]string, 0)
 	for _, member := range membersInfo.Members {
+                // corresponds to https://github.com/tikv/pd/blob/43baea981b406df26cd49e8b99cc42354f0a6696/server/join/join.go#L88.
+                // When multi-cluster enabled, the PD member name is not pod name(cluster1-pd-0) but the FQDN (cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local).
```
Comment (dragonly) [inline pkg/discovery/discovery.go:144]:
```suggestion
              // corresponds to https://github.com/tikv/pd/blob/43baea981b406df26cd49e8b99cc42354f0a6696/server/join/join.go#L88.
              // When multi-cluster enabled, the PD member name is not pod name(cluster1-pd-0) but the FQDN (cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local).
```

File: pkg/discovery/discovery.go (hunk 1 above)
Comment (dragonly) [reply pkg/discovery/discovery.go:149]: We should leave comments as is done in the kubernetes project...

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

File: pkg/discovery/discovery.go [hunk 4]
```diff
@@ -143,10 +143,16 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 
 	membersArr := make([]string, 0)
 	for _, member := range membersInfo.Members {
+		// Corresponds to https://github.com/tikv/pd/blob/43baea981b406df26cd49e8b99cc42354f0a6696/server/join/join.go#L88.
+		// When multi-cluster enabled, the PD member name is not pod name(cluster1-pd-0) but the FQDN (cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local).
+		// For example,
+		// advertisePeerURL without cluster domain: strArr = cluster1-pd-0:2379, member.Name: cluster1-pd-0:2379
```
Comment (DanielZhangQD) [inline pkg/discovery/discovery.go:149]:
```suggestion
		// advertisePeerURL without cluster domain: strArr = cluster1-pd-0.cluster1-pd-peer.pingcap.svc:2380, member.Name: cluster1-pd-0
```

File: pkg/discovery/discovery.go [hunk 5]
```diff
@@ -143,10 +143,16 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
...
+		// advertisePeerURL with cluster domain: strArr = cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local:2379, member.Name: cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local
```
Comment (DanielZhangQD) [inline pkg/discovery/discovery.go:150]:
```suggestion
		// advertisePeerURL with cluster domain: strArr = cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local:2380, member.Name: cluster1-pd-0.cluster1-pd-peer.pingcap.svc.cluster.local
```

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Comment by dragonly** [comment]
LGTM

**Review by dragonly — APPROVED** [review]

**Review by lonng — APPROVED** [review]
LGTM

**Pull request merged**